Readme


										########################
										##					          ##
										##	  	KASSBRIK      ##
										##                    ##
										########################









	Kassbrik est une modeste version du célèbre jeu du Casse-briques. Ce projet a était réaliser dans le cadre de la spécialité scientifique ISN au LycÃ©e Balzac. Il est l'aboutissement d'une annÃ©e de spÃ©cialitÃ© pour notre groupe composÃ© de 3 personnes. Nous avons unifiÃ© nos connaissances afin de pouvoir monter ce projet. 

	Kassbrik est un programme écrit pour fonctionner avec Python 2.7.3 et son module Pygame. 

	Les graphismes, minimalistes, ont Ã©tÃ© rÃ©alisÃ©s avec GIMP.



##########################
##                      ##
##    MATERIEL REQUIS   ##
##                      ##
##########################

	- SystÃ¨me d'exploitation : UBUNTU 

	- Installer Python 2.7.3 sur le terminal

		1)	TÃ©lÃ©charger Python 2.7.3 pour Ubuntu 
		2)	Depuis le terminal, rejoindre le dossier Python2.7.3
		3)	Entrer "./configure" dans le terminal
		4) 	Entrer "make" dans le terminal
		5)	Enfin, entrer "sudo make install" dans le terminal

				Python est installÃ©
	
	- Installer le module Pygame pour Python 2.7.3 en entrant dans le terminal "sudo apt-get install python-pygame" 


##########################
##		                ##
##   LANCEMENT DU JEU   ##
##		                ##
##########################
  
  1)	Depuis le terminal, rejoindre le dossier contenant Kassbrik.py
  
  2)	Puis entrer "python Kassbrik.py" dans le terminal

	
		Le jeu est lancÃ©

##########################
##                      ##
##       COMMANDES      ##
##                      ##
##########################

	Pour dÃ©placer la raquette vers la gauche  : presser le bouton GAUCHE du clavier

	Pour dÃ©placer la raquette vers la droite : presser le bouton DROITE du clavier

	Pour rÃ©demarrer une partie lors de l'affichage de GAME OVER : presser le bouton ESPACE du clavier

	Pour quitter le jeu : cliquer sur la CROIX ROUGE de la fenÃªtre
























Kassbrik/Changelog
										
										CHANGELOG




14/02/2013 : Création du projet Kassbrik, recherches internet pour se
familiariser avec Pygame, le module de Python qui nous permettra de former
notre jeu, et distribution
travail

21/02/2013 : Images de bases créees (raquette, balle, fond d'écran), ouverture
et fermeture de la fenêtre de jeu et affichage des images sur la fenêtre de
jeu

07/03/2013 : Mouvement d'un objet sur un fond, la raquette peut désormais se
déplacer

14/03/2013 : Nous avons aujourd'hui réussi à placer la raquette en bas de
l'écran dès l'ouverture du jeu. Elle se déplace de gauche à droite et ne sort
pas de l'écran. L'image de la balle ne fonctionne pas, une nouvelle a donc été
créée.

15/03/2013 : La balle s'affiche et bouge uniquement vers le bas
sans rebondir nul part.

28/03/2013 : Aujourd'hui, la balle rebondit sur la raquette. Si elle touche
les bouts de la raquette, son angle en rebondissant sera plus conséquent que
si elle touche la raquette en son milieu. De plus la balle rebondit sur les
parois de la fenêtre d'après la loi de Descartes c'est-à-dire que l'angle avec
lequel elle frappe la paroi sera le même après le rebond.

04/04/2013 : Nous avons mis en commun les programmes concernant la
collision-disparition des briques et concernant les rebonds de la balle sur
les parois et sur la raquette. Désormais, lorsque le joueur perd la balle,
"GAMEOVER" s'affiche à l'écran.

25/04/2013 : Nous avons pofiné notre programme pour une meilleure lisibilité
et nous avons fini d'écrire les lignes de code pour détruire les briques
lorsque la balle touche un de leurs côtés. Cependant quelques bugs persistent
car la raquette sort de l'écran et la balle passe à travers certaines briques.

02/05/2013 : Le programme a été entièrement débugué et à nouveau pofiné,
nous envisageons pour le prochain et dernier cours de créer une fonction
permettant de gagner puisque qu'à l'heure actuelle il est impossible de
gagner, le jeu ne s'arrêtant pas lorsque toutes les briques ont disparu.

02/05/2013 : 


Kassbrik/sandbox/testmouv.py
#!/usr/bin/python
# -*- coding: utf8 -*-

import pygame
from pygame.locals import *

RAQUETTE = "../src/img/raquette1.jpeg"
BALLE = "../src/img/balle.png"


pygame.init()
fenetre = pygame.display.set_mode((480, 640))


#affiche le fond
fond = pygame.image.load("../src/img/fondecranbleu.jpg").convert() 
fenetre.blit(fond, (0,0))

#affiche le perso 
raquette = pygame.image.load(RAQUETTE).convert_alpha()
position_raquette = raquette.get_rect()
position_raquette = fenetre.blit(raquette, (180,600))


#flip font pour rafraichir
fenetre.blit(raquette, position_raquette)
pygame.display.flip()

#affiche la balle
ball = pygame.image.load(BALLE).convert_alpha()
position_ball = ball.get_rect()
postion_ball = fenetre.blit(ball, (220,300))



#flip font pour rafraichir
pygame.display.flip()


pygame.key.set_repeat(1, 1)
continuer = 1
while continuer:
    # print dir(position_perso)
   # print position_raquette.x, ',', position_raquette.y
    for event in pygame.event.get():#on parcourt la liste des event de la boucle 
        if event.type == QUIT:
            continuer = 0
        if event.type == KEYDOWN: 
            if event.key == K_LEFT and position_raquette.x > 000: #si fleche gauche et bloquage de la raquette sur la gauche au bord de la fenetre 
                position_perso = position_raquette.move(-5,0)
        if event.type == KEYDOWN:
            if event.key == K_RIGHT and position_raquette.x < 380: #si la fleche droite et bloquage raquette  sur la droite au bord de la fenetre
                position_perso = position_raquette.move(5,0)

#deplacement balle
    position_ball = position_ball.move(0,5)


    #Re- collage
    fenetre.blit(fond,(0,0))
    fenetre.blit(raquette, position_raquette)
    fenetre.blit(ball, position_ball)

    #rafraichissement
    pygame.display.flip()


Kassbrik/sandbox/testcollision5.py
#!/usr/bin/python
# -*- coding: utf8 -*-

import sys
import math
import pygame
from pygame.locals import *

RAQUETTE = "../src/img/raquette1.jpeg"
BALLE = "../src/img/balle2.png"
BRIQUE = "../src/img/brique.jpeg"
GAMEOVER = "../src/img/gameover.png"

vie = 3 
pygame.init()
fenetre = pygame.display.set_mode((480, 640))

dx = 0
dy = 5
dz = 0

#affiche le fond
fond = pygame.image.load("../src/img/fondecranbleu.jpg").convert()
fenetre.blit(fond, (0,0))

#affiche le raquette 
raquette = pygame.image.load(RAQUETTE).convert_alpha()
pos_raquette = raquette.get_rect()
pos_raquette = fenetre.blit(raquette, (180,600))

ball = pygame.image.load(BALLE).convert_alpha()
pos_ball = ball.get_rect()
pos_ball = fenetre.blit(ball, (220,300+dy))

pos_briques = [(200, 150), (200,100), (200,50)]

briques = list()
for pos in pos_briques :
    brik = pygame.image.load(BRIQUE).convert_alpha()
    briques.append(brik)
    fenetre.blit(brik, pos)


#flip font pour rafraichir
pygame.display.flip()

pygame.key.set_repeat(1, 1)
continuer = 1
while continuer == 1:
    for event in pygame.event.get():#on parcourt la liste des évènements pygame 
        if event.type == QUIT:
            continuer = 0
        print event.type
        if event.type == KEYDOWN:
            if event.key == K_LEFT and pos_raquette.x > 000 : #si fleche gauche
                dz = -5
                print pos_raquette.x
            if event.key == K_RIGHT and pos_raquette.x < 380 : #si la fleche droite
                dz = 5
                print pos_raquette.x
            pos_raquette = pos_raquette.move(dz,0)
   
    
    if pos_ball.y == pos_raquette.y and pos_ball.x >= pos_raquette.x+40 and pos_ball.x <= pos_raquette.x+60 :
        dy = -1.05*dy
        dx = 0
    if pos_ball.y == pos_raquette.y and pos_ball.x >= pos_raquette.x+20 and pos_ball.x < pos_raquette.x+40 :
        dy = -1.05*dy
        dx = -1
    if pos_ball.y == pos_raquette.y and pos_ball.x >= pos_raquette.x and pos_ball.x < pos_raquette.x+20 :
        dy = -1.05*dy
        dx = -3
    if pos_ball.y == pos_raquette.y and pos_ball.x > pos_raquette.x+60 and pos_ball.x <= pos_raquette.x+80 :
        dy = -1.05*dy
        dx = 1
    if pos_ball.y == pos_raquette.y and pos_ball.x > pos_raquette.x+80 and pos_ball.x <= pos_raquette.x+100 :
        dy = -1.05*dy
        dx = 3

    if pos_ball.x <=0 or pos_ball.x >= 470 :
        dx = -1.05*dx
        dy = dy

    if pos_ball.y == 620 :
        dx = 1*dx
        dy = -1*dy
     #   continuer = 2 

    if pos_ball.y == 0 :
        dy = -1*dy 

    pos_ball = pos_ball.move(dx,dy)    
   # pos_raquette = pos_raquette.move(dx,0)

#le sytéme de vie 
    if pos_ball.y == 620 : 
        vie = vie -1
        if vie == 0 : 
            continuer = 2


    for brik in briques :
        pos = pos_briques[briques.index(brik)]
        if pos_ball.y == pos[1]+40 :
            dy = -1*dy
            # je supprime la brique de la liste
            pos_briques.remove(pos)
            briques.remove(brik)

   
    #Re- collage
    fenetre.blit(fond, (0,0))
    for brik in briques :
        pos = pos_briques[briques.index(brik)]
        fenetre.blit(brik, pos)

    fenetre.blit(raquette, pos_raquette)
    fenetre.blit(ball, pos_ball)
    #rafraichissement
    pygame.display.flip()

while continuer == 2:
    gameover = pygame.image.load(GAMEOVER).convert_alpha()
    pos_gameover = gameover.get_rect()
    pos_gameover = fenetre.blit(gameover, (0,0))
    fenetre.blit(gameover, pos_gameover)
    pygame.display.flip()
    
    for event in pygame.event.get():
        if event.type == QUIT:
                continuer = 0
        


Kassbrik/sandbox/explosion_brique.py
#!/usr/bin/python
# -*- coding: utf8 -*-

#bibliotheque
import pygame
from pygame.locals import *

#constantes
RAQUETTE = "../src/img/raquette1.jpeg"
BALLE = "../src/img/balle2.png"
BRIQUE = "../src/img/brique.jpeg"
GAMEOVER = "../src/img/gameover.png"


pygame.init()
fenetre = pygame.display.set_mode((480, 640))

#deplacement sur x, y et z
dx = 0
dy = 5
dz = 0

#affiche le fond
fond = pygame.image.load("../src/img/fondecranbleu.jpg").convert()
fenetre.blit(fond, (0,0))

#affiche le raquette 
raquette = pygame.image.load(RAQUETTE).convert_alpha()
pos_raquette = raquette.get_rect()
pos_raquette = fenetre.blit(raquette, (180,600))

#affiche la balle
ball = pygame.image.load(BALLE).convert_alpha()
pos_ball = ball.get_rect()
pos_ball = fenetre.blit(ball, (220,300+dy))

#affiche les briques
pos_briques = [(200, 150), (200,100), (200,50)]
briques = list()
for pos in pos_briques :
    brik = pygame.image.load(BRIQUE).convert_alpha()
    briques.append(brik)
    fenetre.blit(brik, pos)


#rafraichissement de l'image
pygame.display.flip()

#deplacement de la raquette
pygame.key.set_repeat(1, 1)
continuer = 1
while continuer == 1 :
    for event in pygame.event.get():#on parcourt la liste des évènements pygame 
        if event.type == QUIT:
            continuer = 0
        if event.type == KEYDOWN:
            if event.key == K_LEFT and pos_raquette.x > 000 : #si fleche gauche
                dz = -5
            if event.key == K_RIGHT and pos_raquette.x < 380 : #si la fleche droite
                dz = 5
            pos_raquette = pos_raquette.move(dz,0)

#deplacement de la balle
    if pos_ball.y == pos_raquette.y and pos_ball.x >= pos_raquette.x+40 and pos_ball.x <= pos_raquette.x+60 :
        dy = -1*dy
        dx = 0
    if pos_ball.y == pos_raquette.y and pos_ball.x >= pos_raquette.x+20 and pos_ball.x < pos_raquette.x+40 :
        dy = -1*dy
        dx = -1
    if pos_ball.y == pos_raquette.y and pos_ball.x >= pos_raquette.x and pos_ball.x < pos_raquette.x+20 :
        dy = -1*dy
        dx = -3
    if pos_ball.y == pos_raquette.y and pos_ball.x > pos_raquette.x+60 and pos_ball.x <= pos_raquette.x+80 :
        dy = -1*dy
        dx = 1
    if pos_ball.y == pos_raquette.y and pos_ball.x > pos_raquette.x+80 and pos_ball.x <= pos_raquette.x+100 :
        dy = -1*dy
        dx = 3

    if pos_ball.x <=0 or pos_ball.x >= 470 :
        dx = -1*dx
        dy = dy

    if pos_ball.y == 620 :
        dx = 0
        dy = 0
        continuer = 2

    if pos_ball.y == 0 :
        dy = -1*dy 

    pos_ball = pos_ball.move(dx,dy)    

#explosion des briques quand la balle vient les toucher
    for brik in briques :
        pos = pos_briques[briques.index(brik)]
        if pos_ball.y-40 == pos[0][1] and pos[0][0] >= pos_ball.x and pos[0][0] <= pos_ball.x+60 :
            dy = -1*dy
#je supprime la brique de la liste
            pos_briques.remove(pos)
            briques.remove(brik)

   
    fenetre.blit(fond, (0,0))
    for brik in briques :
        pos = pos_briques[briques.index(brik)]
        fenetre.blit(brik, pos)

    fenetre.blit(raquette, pos_raquette)
    fenetre.blit(ball, pos_ball)

    #rafraichissement
    pygame.display.flip()

while continuer == 2:
    gameover = pygame.image.load(GAMEOVER).convert_alpha()
    pos_gameover = gameover.get_rect()
    pos_gameover = fenetre.blit(gameover, (0,0))
    fenetre.blit(gameover, pos_gameover)
    pygame.display.flip()

    for event in pygame.event.get():
        if event.type == QUIT:
                continuer = 0






Kassbrik/sandbox/explosion_brique2.py
#!/usr/bin/python
# -*- coding: utf8 -*-

import sys
import math
import pygame
from pygame.locals import *

RAQUETTE = "../src/img/raquette1.jpeg"
BALLE = "../src/img/balle2.png"
BRIQUE = "../src/img/brique.jpeg"
GAMEOVER = "../src/img/gameover.png"


pygame.init()
fenetre = pygame.display.set_mode((480, 640))

dx = 0
dy = 5
dz = 0

#affiche le fond
fond = pygame.image.load("../src/img/fondecranbleu.jpg").convert()
fenetre.blit(fond, (0,0))

#affiche le raquette 
raquette = pygame.image.load(RAQUETTE).convert_alpha()
pos_raquette = raquette.get_rect()
pos_raquette = fenetre.blit(raquette, (180,600))

ball = pygame.image.load(BALLE).convert_alpha()
pos_ball = ball.get_rect()
pos_ball = fenetre.blit(ball, (220,300+dy))

pos_briques = [(200, 150), (200,100), (200,50)]

briques = list()
for pos in pos_briques :
    brik = pygame.image.load(BRIQUE).convert_alpha()
    briques.append(brik)
    fenetre.blit(brik, pos)


#flip font pour rafraichir
pygame.display.flip()

pygame.key.set_repeat(1, 1)
continuer = 1
while continuer:
    for event in pygame.event.get():#on parcourt la liste des évènements pygame 
        if event.type == QUIT:
            continuer = 0
        if event.type == KEYDOWN:
            if event.key == K_LEFT and pos_raquette.x > 000 : #si fleche gauche
                dz = -5
            if event.key == K_RIGHT and pos_raquette.x < 380 : #si la fleche droite
                dz = 5
            pos_raquette = pos_raquette.move(dz,0)
   

    if pos_ball.y == pos_raquette.y and pos_ball.x >= pos_raquette.x+40 and pos_ball.x <= pos_raquette.x+60 :
        dy = -1*dy
        dx = 0
    if pos_ball.y == pos_raquette.y and pos_ball.x >= pos_raquette.x+20 and pos_ball.x < pos_raquette.x+40 :
        dy = -1*dy
        dx = -1
    if pos_ball.y == pos_raquette.y and pos_ball.x >= pos_raquette.x and pos_ball.x < pos_raquette.x+20 :
        dy = -1*dy
        dx = -3
    if pos_ball.y == pos_raquette.y and pos_ball.x > pos_raquette.x+60 and pos_ball.x <= pos_raquette.x+80 :
        dy = -1*dy
        dx = 1
    if pos_ball.y == pos_raquette.y and pos_ball.x > pos_raquette.x+80 and pos_ball.x <= pos_raquette.x+100 :
        dy = -1*dy
        dx = 3

    if pos_ball.x <=0 or pos_ball.x >= 470 :
        dx = -1*dx
        dy = dy

    if pos_ball.y == 620 :
        dx = 0
        dy = 0
        gameover = pygame.image.load(GAMEOVER).convert_alpha()
        pos_gameover = gameover.get_rect()
        pos_gameover = fenetre.blit(gameover, (130,215))      
        fenetre.blit(gameover, pos_gameover)

    if pos_ball.y == 0 :
        dy = -1*dy 

    pos_ball = pos_ball.move(dx,dy)    
   # pos_raquette = pos_raquette.move(dx,0)

    for brik in briques :
        pos = pos_briques[briques.index(brik)]
        if pos_ball.y == pos[1]+40 :
            dy = -1*dy
            # je supprime la brique de la liste
            pos_briques.remove(pos)
            briques.remove(brik)

   
    #Re- collage
    fenetre.blit(fond, (0,0))
    for brik in briques :
        pos = pos_briques[briques.index(brik)]
        fenetre.blit(brik, pos)

    fenetre.blit(raquette, pos_raquette)
    fenetre.blit(ball, pos_ball)
    #rafraichissement
    pygame.display.flip()



Kassbrik/sandbox/fairebougerrectangle.py
#!/usr/bin/python
# -*- coding: utf8 -*-

import pygame
from pygame.locals import *

#initialise l'affichage 
pygame.display.init() 


#cree la resolution et affiche l'affichage
RESOLUTION = (640 , 480)
screen = pygame.display.set_mode(RESOLUTION)

#definit la couleur et l'affiche
couleur = 0, 0, 0
screen.fill(couleur)

rectangle=pygame.Surface((60,10)) #definit la surface du rectangle
rectangle.fill ( (255,0,0) ) #definit la couleur du rectangle
screen.blit(rectangle, (260,485)) #definit la position du rectangle(plaque l'objet sur la surface d'affichage)

pygame.key.set_repeat(40, 30)

position_rectangle=rectangle.get_rect()

#boucle qui permet le mouvement de l'image
continuer = 1
while continuer:
    for event in pygame.event.get(): #permet le parcours la liste des event de la boucle 
        if event.type == QUIT: #boucle infinie pour que la fenetre ne se ferme pas toute seule mais grace à la croix
            continuer = 0

        if event.type == KEYDOWN:
            if event.key == K_LEFT: #si fleche gauche
                position_rectangle = position_rectangle.move(-3,0) #permet de deplacer l'objet vers la gauche
            if event.key == K_RIGHT: #si la fleche droite
                position_rectangle = position_rectangle.move(3,0) #permet de deplacer l'objet vers la droite
    
    #FIXME trouver comment effacer le rectangle avant de le reafficher 
    # plus loin

    #actualise l'affichage
    pygame.display.flip()
    



Kassbrik/sandbox/testcollision.py
#!/usr/bin/python
# -*- coding: utf8 -*-

import sys
import math
import pygame
from pygame.locals import *

RAQUETTE = "../src/img/raquette1.jpeg"
BALLE = "../src/img/balle2.png"
BRIQUE = "../src/img/brique.jpeg"
GAMEOVER = "../src/img/gameover.png"

pygame.init()
fenetre= pygame.display.set_mode((480, 640))


dx = 6
dy = 6
Dx = 0

#affiche le fond
fond = pygame.image.load("../src/img/fondecranbleu.jpg").convert()
fenetre.blit(fond, (0,0))

#affiche les briques

pos_briques = [(200, 150), (200,100), (200,50)]

briques = list()
for pos in pos_briques :
    brik = pygame.image.load(BRIQUE).convert_alpha()
    briques.append(brik)
    fenetre.blit(brik, pos)


#affiche le raquette 
raquette = pygame.image.load(RAQUETTE).convert_alpha()
pos_raquette = raquette.get_rect()
pos_raquette = fenetre.blit(raquette, (180,600))


#affiche la balle
ball = pygame.image.load(BALLE).convert_alpha()
pos_ball = ball.get_rect()
pos_ball = fenetre.blit(ball, (220,300+dy))



#flip font pour rafraichir
pygame.display.flip()


pygame.key.set_repeat(1, 1)

continuer = 1
while continuer == 1 :
    for event in pygame.event.get():#on parcourt la liste des évènements pygame 
        if event.type == QUIT:
            continuer = 0
        if event.type == KEYDOWN:
            if event.key == K_LEFT and pos_raquette.x > 000 : #si fleche gauche
                Dx = -5 
                pos_raquette = pos_raquette.move(Dx,0)
            if event.key == K_RIGHT and pos_raquette.x < 380 : #si la fleche droite
                Dx = 5
                pos_raquette = pos_raquette.move(Dx,0)
   

 #les lignes suivantes modifie l'angle de la trajectoire de la balle selon l'endroit ou elle tape la raquette 
    if pos_ball.y == pos_raquette.y and pos_ball.x >= pos_raquette.x+40 and pos_ball.x <= pos_raquette.x+60 :
        dy = -1*dy
        dx = 0
    if pos_ball.y == pos_raquette.y and pos_ball.x >= pos_raquette.x+20 and pos_ball.x < pos_raquette.x+40 :
        dy = -1*dy
        dx = -1
    if pos_ball.y == pos_raquette.y and pos_ball.x >= pos_raquette.x and pos_ball.x < pos_raquette.x+20 :
        dy = -1*dy
        dx = -3
    if pos_ball.y == pos_raquette.y and pos_ball.x > pos_raquette.x+60 and pos_ball.x <= pos_raquette.x+80 :
        dy = -1*dy
        dx = 1
    if pos_ball.y == pos_raquette.y and pos_ball.x > pos_raquette.x+80 and pos_ball.x <= pos_raquette.x+100 :
        dy = -1*dy
        dx = 3

#si touche en haut de l'ecran
    if pos_ball.y == 0 :
        dy = -1*dy
        dx = 1*dx

#si touche a droite ou a gauche de l'ecran
    if pos_ball.x <= 0 or pos_ball.x >= 470:
        dy = 1*dy
        dx = -1*dx

#si touche en bas
    if pos_ball.y == 620 :
        dx = 0
        dy = 0
        continuer == 2

    for brik in briques :
        pos = pos_briques[briques.index(brik)]
        if pos_ball.y == pos[1]+40 :
            dy = -1*dy
            # je supprime la brique de la liste
            pos_briques.remove(pos)
            briques.remove(brik)


    pos_ball = pos_ball.move(dx,dy)
   # pos_raquette = pos_raquette.move(dx,0)

    #Re- collage
    fenetre.blit(fond, (0,0))
    for brik in briques :
        pos = pos_briques[briques.index(brik)]
        fenetre.blit(brik, pos)
        fenetre.blit(raquette, pos_raquette)
        fenetre.blit(ball, pos_ball)
   
while continuer == 2:
    gameover = pygame.image.load(GAMEOVER).convert_alpha()
    pos_gameover = gameover.get_rect()
    pos_gameover = fenetre.blit(gameover, (0,0))
    fenetre.blit(gameover, pos_gameover)
    pygame.display.flip()

    for event in pygame.event.get():
        if event.type == QUIT:
                continuer = 0


    #rafraichissement
    pygame.display.flip()

